cmd/run/run_test.go

```diff
@@ -378,6 +379,43 @@ func TestBuildServiceWithPresharedKeyAuthentication(t *testing.T) {
 	}
 }
 
+func TestBuildServiceWithTracingEnabled(t *testing.T) {
+	// create mock OTLP server
+	otlpServerPort, otlpServerPortReleaser := TCPRandomPort()
+	localOTLPServerURL := fmt.Sprintf("localhost:%d", otlpServerPort)
+	otlpServerPortReleaser()
+	otlpServer, err := mocks.NewMockTracingServer(otlpServerPort)
```

Comment:
You need a way to close the mock server, otherwise we leak in our tests.

There should be a `defer otlpServer.Close()` that closes the underlying grpc server.